Std/Logic/Function.lean

```diff
@@ -0,0 +1,264 @@
+/-
+Copyright (c) 2014 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Jeremy Avigad, Haitao Zhang, Yury Kudryashov
+-/
+import Std.Logic.Basic
+
+/-!
+# General operations on functions
+
+In this file we define the following predicates on functions `f : α → β`:
+
+- `Function.Injective`: `f x = f y` implies `x = y`;
+- `Function.Surjective`: each `y` is equal to `f x` for some `x`;
+- `Function.Bijective`: a function is both injective and bijective;
+- `Function.Involutive`: `f (f x) = x` for all `x`.
+
+We also prove some theorems about these definitions and functions in general.
+-/
+
+set_option autoImplicit false
+
```

Comment:
I think usually the approach at Std is the opposite, to take advantage of auto implicits where possible.

Reply:
This was temporary, forgot to remove.